lib/galaxy/exceptions/__init__.py

```diff
@@ -103,6 +103,11 @@ class ToolMissingException(MessageException):
     status_code = 400
     err_code = error_codes.USER_TOOL_MISSING_PROBLEM
 
+    @property
+    def tool_id(self):
+        if 'tool_id' in self.extra_error_info:
+            return self.extra_error_info['tool_id']
```

Comment:
What about overloading `__init__` instead?

```python
    def __init__(self, err_msg=None, type="info", tool_id=None, **extra_error_info):
        super(ToolMissingException, self).__init__(err_msg, type, **extra_error_info)
        self.tool_id = tool_id
```

Reply:
Sure, that works for me.